DEVELOPMENT.md

```diff
@@ -250,6 +250,78 @@ targets in favor of the Python-specific make targets:
    `/usr/local/lib/pgai`.
 
 
+#### Building prerelease versions of the extension
+
+We would like to avoid long-lived feature branches and the git headaches that entails. 
+Yet some features take a while to mature to a level that we are happy to ship and 
+support in production environments. Avoiding feature branches means we may be in 
+a position where we need to ship but have SQL code that is not fully ready. 
+Making changes to SQL objects once they ship is tricky and painful and should be 
+minimized.
+
+We want to:
+- avoid long-lived feature branches. 
+- merge progress to main
+- minimize database migrations that alter existing objects
+- avoid installing prerelease code in production databases
+- allow prerelease code to be installed/tested/trialed in an opt-in manner
+- warn when prerelease code is installed
+- avoid supporting extension upgrades after prerelease code has been installed
+
+To this end, we allow prerelease functionality to be gated behind feature flags.
+We only build and ship prerelease code if the version being built includes a 
+prerelease tag. If the version is a production version, we omit the prerelease 
+code altogether. Furthermore, if the feature flag is not enabled, the prerelease 
+code is NOT executed/installed. 
+
+Incremental or idempotent files numbered greater than 899 must have a comment at 
```

Comment:
Your implementation only takes 3 digits. I know we won't be potentially having such amount of files but I believe better to mention it here.
```suggestion
Incremental or idempotent files numbered greater than 899 and lower than 999 must have a comment at 
```